### PR TITLE
fix(attack-path): resolve a payload's real target instead of its payload default (#7231)

### LINKED ISSUE #7231: fix(attack-path): payloads attach to a shared 'localhost' node, and NetworkTraffic payloads never appear
Two defects in the attack-path target resolution leave the map misrepresenting what ran where. Both live in `AttackPathExecutionIngestionService`, in the payload-type switch that turns an executed inject into `attackpath_execution` rows.

## 1. A payload can be attached to a "localhost" node instead of its endpoint

For a `COMMAND` payload, the target is taken from the payload's **argument default value**:

```java
List<String> targetArg = payloadCommand.getArguments().stream()
    .filter(arg -> typeEndpoint.contains(arg.getType()))
    .map(PayloadArgument::getDefaultValue)   // <-- the default, not the value this inject uses
    .toList();
if (targetArg.size() == 1) targetArgIdentified = targetArg.getFirst();
```

Two consequences:

- **The default wins over the value actually used.** The default is authored once on the payload; the value is chosen per inject. An inject overriding `target_ip` still lands on the payload's default.
- **A loopback value becomes a discovered endpoint.** A `host`-typed argument defaulting to `localhost` is extremely common — a payload in our own dataset (`echo seb`) has exactly one endpoint-typed argument, `host`, defaulting to `localhost`. The command never left the machine it ran on, yet the graph creates a `DISCOVERED` node literally named `localhost`. Because the target key is the raw value, **every endpoint's local executions collapse into that one shared node**, and the endpoint that actually ran the payload disappears from the chain.

## 2. `NetworkTraffic` payloads never appear at all

The switch handles `EXECUTABLE`, `FILE_DROP`, `AI_ATTACK`, `DNS_RESOLUTION` and `COMMAND` — but `PayloadType` also has **`NETWORK_TRAFFIC`**. It is a `switch` *statement*, so the missing branch compiles silently and writes **zero rows**: the action never lands on the attack path. The endpoint is still drawn (other executions target it), which is exactly the reported symptom — endpoints with no payload card.

`getExecutionIndex` duplicates the same switch and so carries both defects identically; the two must agree, since it recomputes the id that the write persisted.

## Fix

- Resolve a Command's target from the **inject's own argument value**, falling back to the payload default, and treat loopback values (`localhost`, `127.0.0.1`, `::1`, `0.0.0.0`, `*`) as "the endpoint it ran on" rather than a discovered target.
- Add the `NETWORK_TRAFFIC` branch, targeting its `ip_dst` (mirroring how `DNS_RESOLUTION` targets its hostname) and falling back to the executing endpoint when the destination is absent or loopback.
- Add a `default` branch to both switches so a payload type added to the enum can never again silently write nothing.
- Share one helper between the write path and `getExecutionIndex` so their ids cannot drift.

### PR DESCRIPTION
**Pull request by Seb-MIGUEL**
Fixes #7231 (reported as OpenAEV-Platform/filigran-private#259 and OpenAEV-Platform/filigran-private#260 - same root area, one switch).

### 1. Payloads attached to a shared `localhost` node instead of their endpoint

`AttackPathExecutionIngestionService` resolved a `COMMAND` payload's target from the argument's **default value** rather than the value the inject actually runs with. So an inject overriding `target_ip` still landed on the payload default, and - far more visibly - a `host` argument defaulting to `localhost` produced a `DISCOVERED` node literally named `localhost`. The target key *is* that raw value, so every endpoint's local executions collapsed onto one shared node and the endpoint that ran the payload disappeared from the chain.

This is not hypothetical: a payload in the local dataset (`echo seb`) has exactly one endpoint-typed argument - `host`, defaulting to `localhost`.

Now: read the inject's own value, fall back to the payload default, and treat loopback (`localhost`, `127.0.0.1`, `::1`, `0.0.0.0`, `*`) as "the endpoint it ran on".

### 2. `NetworkTraffic` payloads never appeared at all

`PayloadType` has six values; the resolution switch handled five. It is a `switch` **statement**, so the missing `NETWORK_TRAFFIC` branch compiled silently and wrote **zero rows** - the action never landed on the map, while its endpoint was still drawn by other executions. That is exactly the reported "endpoints with no payload card". Verified against the current tip of `main`: `PayloadType.NETWORK_TRAFFIC` and the `NetworkTraffic` entity are both present and actively used (`PayloadService`, `InjectUtils`, `PayloadCreateInput`, frontend type mappings), so the gap is real today.

Now: `NETWORK_TRAFFIC` targets its `ip_dst`, mirroring how `DNS_RESOLUTION` targets its hostname, falling back to the executing endpoint when the destination is absent or loopback. Both switches also gained a `default` branch so a payload type added to (or later removed from) the enum can never again silently write nothing.

`getExecutionIndex` duplicated the same resolution and carried both defects identically - it must agree with the write path since it recomputes the persisted id. Both now share one helper.

Review note (#1191): the network-traffic *attack capability* (an executor that generates the traffic) was never implemented - #1191 tracks that feature and is still open - but the payload type is creatable today (`PayloadCreateInput`, `PayloadService`), so an inject carrying one must not silently write zero attack-path rows. If the type is ever removed as part of that feature decision, the `default` branch added here keeps the graph safe through the removal, and dropping the `case NETWORK_TRAFFIC` is a one-line mechanical follow-up. On the "payloads hidden when target == source" design (filigran-private#239): that was the previous design; per the team the new design shows payloads and injectors on the map, which is what the loopback-to-ASSET behavior here implements.

### 3. Review hardening: normalize the target value exactly once

All remote-target resolution (Command argument value, NetworkTraffic `ip_dst`) flows through a single `remoteTargetOrNull` helper: the persisted value / id key is trimmed once, so stray whitespace can never fork a second discovered node (the authored case is kept for display; only the loopback comparison lowercases). A blank or JSON-null inject override falls back to the payload default instead of suppressing it, and a JSON null node no longer reads as the literal string "null". `DNS_RESOLUTION` is deliberately untouched: its hostname is fixed on the payload (not argument-driven), there is no reported defect there, and re-keying it would churn existing DNS node ids for no benefit.

### 4. Review follow-up: targeted-asset arguments resolve like execution does

A `TargetedAsset` inject field is a JSON array of endpoint ids, not a scalar: reading it as text yields blank, which fell back to the payload default (the pre-existing behavior). It now goes through the same resolver execution uses (`InjectService#retrieveValuesOfTargetedAssetFromInject`), so the row carries the property value (hostname / seen IP / local IP) the command actually ran with. The row model keys exactly one row per `(inject, agent)` - `getExecutionIndex` must recompute the persisted id from that pair alone, which is how Phase-B verdicts, traces, and collector snapshots find their row - so a selection resolving to several destinations stays ambiguous (same rule as several endpoint-ish arguments) and falls back to the executing endpoint; fanning out to one row per destination needs the Phase-B index to become multi-valued first, an explicit follow-up. Resolution failures degrade to the executing-endpoint edge rather than failing the run's write.

### Verification

Three integration tests from the original fix, each run against the code **with and without** the fix:

| test | without the fix |
|---|---|
| command whose host argument is loopback | `expected "ASSET" but was "DISCOVERED"` |
| command target from the inject's argument | `expected "192.168.56.11" but was "10.9.9.9"` |
| NetworkTraffic payload is ingested | `Expected size: 1 but was: 0` |

Six review additions: a blank override falls back to the payload default, the persisted target value is trimmed, a loopback NetworkTraffic destination stays on its endpoint, a targeted-asset argument resolves to the selected endpoint's property value, a multi-endpoint selection falls back to the executing endpoint, and the recomputed execution index equals the persisted row id on every touched branch. The tests introduced by this PR follow the `given_X_should_Y` naming convention. Full test class: `Tests run: 18, Failures: 0, Errors: 0`. Branch rebased onto current `main` (clean, no conflicts; all commits signed); Spotless applied.

### CODE REVIEW

**Review by EvaE-Filigran — CHANGES_REQUESTED** [review]
Payload NETWORK_TRAFFIC do not exist anymore in the project reset this PR

File: openaev-api/src/main/java/io/openaev/service/attackpath/ingestion/AttackPathExecutionIngestionService.java [hunk 1]
```diff
@@ -57,6 +57,62 @@ public class AttackPathExecutionIngestionService {
...
+    JsonNode override =
+        inject.getContent() == null ? null : inject.getContent().get(argument.getKey());
+    String value =
+        override != null && !override.asText().isEmpty()
+            ? override.asText()
+            : argument.getDefaultValue();
+    if (value == null || value.isBlank()) {
+      return null;
+    }
+    return SELF_TARGET_VALUES.contains(value.trim().toLowerCase()) ? null : value;
```
[the remark this answers is not in the document]
Comment (SamuelHassine) [reply openaev-api/src/main/java/io/openaev/service/attackpath/ingestion/AttackPathExecutionIngestionService.java]: Fixed in d428fd04f, together with the two suppressed duplicates of the same root issue. All remote-target resolution (Command argument value and NetworkTraffic ip_dst) now flows through a single `remoteTargetOrNull` helper that trims exactly once: the persisted value / id key is the trimmed value (authored case kept for display), and only the self-target comparison lowercases. A whitespace-only override no longer blocks the fallback to the payload default (`isBlank` check on the override), and a JSON null node no longer reads as the literal string "null". Both the write path and `getExecutionIndex` share the helper, so they cannot drift; new tests assert the recomputed index equals the persisted row id and cover the blank-override, trimmed-value, and loopback-destination edge cases.

**Comment by SamuelHassine** [comment]
@EvaE-Filigran Thanks for the review! I double-checked against the current tip of `main` (e706e67b7, fetched today) before touching anything, and NETWORK_TRAFFIC does still exist in the project:

- `openaev-model/src/main/java/io/openaev/database/model/PayloadType.java` declares six values: COMMAND, EXECUTABLE, FILE_DROP, DNS_RESOLUTION, NETWORK_TRAFFIC, AI_ATTACK.
- The `openaev-model/src/main/java/io/openaev/database/model/NetworkTraffic.java` entity is present, with active backend usages on `main`: `PayloadService` (a `case NETWORK_TRAFFIC` branch), `InjectUtils`, and `PayloadCreateInput`.
- The frontend still maps the type too: `PayloadIcon.tsx`, `InjectIcon.tsx`, `api-types.d.ts`, and the translation bundles all carry "NetworkTraffic".
- I found no merged or open PR that removes it (`gh pr list --search "network traffic"`).

So the missing NETWORK_TRAFFIC branch in the ingestion switch is a real gap today: a NetworkTraffic inject writes zero attack-path rows, which is exactly the "endpoint with no payload card" symptom from #7231. If a removal of the payload type is planned, the `default` branch this PR adds to both switches is the piece that protects the graph through that change (any type without a branch falls back to the executing endpoint instead of silently writing nothing), and dropping the `NETWORK_TRAFFIC` case would then be a mechanical follow-up of that removal PR.

Could you take another look and, if you agree, re-review? If there is a removal in flight I missed, happy to be pointed to it and I will rework the PR accordingly.

**Review by SamuelHassine — APPROVED** [review]
Reviewed the full service and test class, not just the diff. The root-cause analysis is right on both counts: the Command target came from the payload-authored default instead of the inject's effective value, and the missing NETWORK_TRAFFIC branch in a switch statement silently wrote zero rows. The shared resolution helper plus the default branches make the write path and getExecutionIndex structurally unable to drift, which is the property that actually matters here since the index recomputes persisted ids.

Follow-ups applied on the branch during review: remote-target values are now normalized (trimmed) exactly once in a helper shared by Command and NetworkTraffic resolution, a blank or JSON-null inject override falls back to the payload default instead of suppressing it, and four tests were added covering the edge cases and asserting index-vs-write id agreement. Branch rebased onto current main with the merge commit flattened; all commits signed. Full test class green locally: 16/16.

Note on the NETWORK_TRAFFIC objection: verified against the current tip of main - PayloadType.NETWORK_TRAFFIC and the NetworkTraffic entity are present and actively used (backend and frontend), so the ingestion gap this PR closes is real; evidence posted in the conversation.

praise: the PR body's with/without-fix verification table and the intent-level comments on the switch default branches are exactly the level of documentation this ingestion code needs.

**Comment by SamuelHassine** [comment]
## Review and fix summary

- **Independent review**: read `AttackPathExecutionIngestionService` and its test class end to end. The fix is correct on both counts (Command target from the inject's effective value, missing NETWORK_TRAFFIC branch writing zero rows); hardening applied on top (see below). Tenant scoping untouched (no new repository lookups). `DNS_RESOLUTION` deliberately left as-is: its hostname is fixed on the payload, no reported defect, and re-keying would churn existing node ids.
- **Copilot review (visible thread + 2 suppressed duplicates)**: fixed in d428fd0. One shared `remoteTargetOrNull` helper now trims the persisted value exactly once (authored case kept for display; only the loopback comparison lowercases), a whitespace-only or JSON-null override falls back to the payload default, and a JSON null node no longer reads as the literal string "null". Id-stability tradeoff: recomputed indexes and persisted rows always live inside one run lifecycle and this PR already re-keys Command targets, so trimming adds no extra migration risk. Thread resolved.
- **NETWORK_TRAFFIC objection (@EvaE-Filigran)**: verified against the current tip of `main` - `PayloadType.NETWORK_TRAFFIC` and the `NetworkTraffic` entity exist and are actively used in backend (`PayloadService`, `InjectUtils`, `PayloadCreateInput`) and frontend (type mappings, icons, translations); no merged or open PR removes them. Evidence posted in the conversation above. **The changes-requested review is the only remaining merge blocker and needs Eva's re-review to lift.**
- **Tests**: 4 added (blank override fallback, trimmed persisted value, loopback NetworkTraffic destination, index-vs-write id agreement), directly covering the previously unexercised `getExecutionIndex` branches flagged by codecov. Full class green locally (16/16) and in CI.
- **Branch hygiene**: rebased onto current `main`, the stale merge commit flattened, original fix commit content-identical (range-diff verified), all commits signed, pushed with `--force-with-lease`.
- **CI**: fully green on d428fd0 including the required `pipeline / Pipeline Gate`. The earlier red `Merge Platforms (ubi9)` and the missing pipeline run on the rebased head were both collateral of yesterday's GitHub Actions incident (run retriggered via close/reopen), not this PR.

**Comment by EvaE-Filigran** [comment]
Network trafic is an oaev payload never implemented : https://github.com/OpenAEV-Platform/openaev/issues/1191; 
The issue report https://github.com/OpenAEV-Platform/filigran-private/issues/259 is not an issue, some payload are not display if the target and the source are the same see issue https://github.com/OpenAEV-Platform/filigran-private/issues/239;

**Comment by mergenhan** [comment]
`some payload are not display if the target and the source are the same` This was for previous decision for the previous desing. with the new design we should show payloads and injectors in the UI.

**Comment by SamuelHassine** [comment]
@EvaE-Filigran Thanks for the pointers - re-checked both against today's tip of main (8b3634f) before responding.

**On #1191 ("never implemented")**: agreed that the network-traffic *attack capability* (an executor that actually generates the traffic) was never built - #1191 is still open. But the payload *type* is fully present and creatable today: `PayloadType.NETWORK_TRAFFIC` (PayloadType.java line 11), the `NetworkTraffic` entity, `PayloadCreateInput`, and a `case NETWORK_TRAFFIC` branch in `PayloadService` all live on main, and the frontend maps the type (icons, api-types, translations). So a user can author a NetworkTraffic payload and attach it to an inject right now; when that inject runs, the ingestion switch used to write zero attack-path rows silently - the "endpoint with no payload card" symptom. The new branch just mirrors what `DNS_RESOLUTION` already does and costs nothing while the executor does not exist. If the type is ever removed as part of resolving or closing #1191, the `default` branch this PR adds is exactly the piece that keeps the graph safe through that removal, and deleting the `case NETWORK_TRAFFIC` becomes a one-line mechanical follow-up of that removal PR.

**On filigran-private#259 vs #239 (payloads hidden when target == source)**: that intentional hiding belonged to the previous design - as @mergenhan confirmed above, the new design should show payloads and injectors on the map. This PR implements exactly that behavior for the self-target case: a command whose host argument is loopback is now attached to the endpoint it ran on (an ASSET edge) instead of either disappearing or collapsing onto a shared "localhost" DISCOVERED node that every endpoint's local runs merged into. Independently of that design question, the second defect fixed here is a plain correctness bug: the target was read from the payload's authored *default* value instead of the value the inject actually runs with, so an inject overriding `target_ip` still landed on the default.

Given the above, could you re-review and lift the changes-requested when you agree? CI is fully green on the current head and the Copilot thread is resolved - your review is the only remaining merge blocker. Happy to rework if there is a removal of the payload type in flight that I could not find.

**Review by SamuelHassine — APPROVED** [review]
Second full review pass (worktree checkout of the head, full service and test class read, not just the diff). No new must-fix or should-fix findings: the shared `resolveCommandTargetValue` / `remoteTargetOrNull` helpers make the write path and `getExecutionIndex` structurally unable to drift, the NETWORK_TRAFFIC branch mirrors DNS_RESOLUTION with a loopback fallback to the executing endpoint, and the `default` branches are documented forward-protection for enum changes (currently unreachable by design since all six values have branches and `PayloadType.fromString` throws on unknown keys).

Re-verified the disputed facts against today's tip of main (8b3634f): `PayloadType.NETWORK_TRAFFIC`, the `NetworkTraffic` entity, and their backend/frontend usages are all present. The open #1191 tracks the never-built executor capability, which does not invalidate handling the creatable payload type defensively in ingestion; detailed response posted in the conversation.

CI fully green on d428fd0 (all pipeline jobs including API tests, E2E matrix, Spotless, CodeQL, signed-commits check). Both commits signed. Copilot thread resolved. Approving; @EvaE-Filigran's changes-requested review remains the only merge blocker and needs her re-review to lift.

**Comment by SamuelHassine** [comment]
## Review and fix summary (second pass, 2026-08-09)

- **Independent review**: fresh full read of `AttackPathExecutionIngestionService` and its test class in a clean worktree of the head. No new must-fix or should-fix findings. The shared resolution helpers keep the write path and `getExecutionIndex` id-consistent by construction, all six `PayloadType` values now have explicit branches, and the `default` branches are documented forward-protection (unreachable today, which is intentional).
- **Code changes this pass**: none needed - the branch is unchanged at d428fd0. CI stayed fully green (pipeline gate, API tests, E2E matrix, Spotless, CodeQL, signed commits), the PR is mergeable/CLEAN, both commits signed.
- **Review threads**: the single Copilot inline thread was already fixed (d428fd0) and resolved; zero unresolved threads remain.
- **NETWORK_TRAFFIC discussion**: replied point-by-point to @EvaE-Filigran's follow-up. Facts re-verified against today's main (8b3634f): the payload type, entity, and creation path all exist; #1191 tracks the never-built executor capability, which does not invalidate defensive ingestion of a creatable type. The target==source hiding was the previous design; per @mergenhan the new design shows payloads on the map, which the loopback-to-ASSET behavior implements. The PR description now carries this context in a review note.
- **Remaining blocker**: the changes-requested review from @EvaE-Filigran is the only thing preventing merge - it needs her re-review (or a maintainer dismissal) to lift. Everything else is done: CI green, threads resolved, title/description conform to the repo convention, approval posted.

File: openaev-api/src/main/java/io/openaev/service/attackpath/ingestion/AttackPathExecutionIngestionService.java [hunk 2]
```diff
@@ -57,6 +57,84 @@ public class AttackPathExecutionIngestionService {
...
 
+  /**
+   * Argument types that can name an endpoint, so a Command payload carrying exactly one of them
+   * targets that value rather than the endpoint it runs on.
+   */
+  private static final Set<PrimitiveType> TARGET_ARGUMENT_TYPES =
+      Set.of(
+          PrimitiveType.IPv4,
+          PrimitiveType.IPv6,
+          PrimitiveType.TargetedAsset,
```
[the remark this answers is not in the document]
Comment (SamuelHassine) [reply openaev-api/src/main/java/io/openaev/service/attackpath/ingestion/AttackPathExecutionIngestionService.java:70]: Fixed in 5245f91ab. A single targeted-asset argument now goes through the same resolver execution uses (InjectService#retrieveValuesOfTargetedAssetFromInject), so the row carries the property value (hostname / seen IP / local IP) the command actually ran with instead of the payload default - the write path and getExecutionIndex share the helper, so their ids still cannot drift. Two integration tests cover it (resolved single destination, and the multi-destination case), both asserting index-vs-write id agreement.

On "account for one argument resolving to multiple destinations": the row model deliberately keys exactly one row per (inject, agent), and getExecutionIndex must be able to recompute that persisted id from (inject, agent) alone - it is how Phase-B verdicts, terminal traces, and collector snapshots find their row. Fanning one argument out to one row per destination would leave N-1 rows unreachable by every Phase-B update until that index becomes multi-valued, so a multi-destination selection is treated as ambiguous under the same pre-existing rule as several endpoint-ish arguments and falls back to the always-true "ran on its endpoint" edge. The helper's javadoc documents this as the follow-up trigger; resolution failures (missing linked targeted-property field, malformed content, deleted endpoints) degrade the same way rather than failing the run's write.

**Review by SamuelHassine — APPROVED** [review]
Third full pass (rebase run, 2026-08-09). Rebased the branch onto today's main (21 commits ahead, zero conflicts; no attack-path, inject-service, or PayloadType changes landed on main since the previous base, so nothing here was superseded). Re-read the full service and test class on the rebased head.

New finding addressed this pass (from Copilot's balanced review, and it was a fair blocking catch): a TargetedAsset argument's inject field is a JSON array of endpoint ids, so the scalar override read yielded blank and fell back to the payload default. It now resolves through the same InjectService#retrieveValuesOfTargetedAssetFromInject execution uses, so the row carries the property value the command actually ran with; a multi-destination selection stays ambiguous under the one-row-per-(inject, agent) model and falls back to the executing endpoint (documented follow-up: multi-valued Phase-B index before fanning out). The write path and getExecutionIndex still share every resolution helper, so their ids cannot drift - both new tests assert that agreement.

Also renamed the tests this PR introduced to the given_X_should_Y convention per testing.instructions.md. Full test class green locally (18/18) and in CI; all three commits signed; CI fully green on 5245f91ab including the pipeline gate, all API test shards, and the E2E matrix.

Approving. @EvaE-Filigran's changes-requested review remains the only merge blocker and needs her re-review to lift.

**Comment by SamuelHassine** [comment]
Review and fix summary (rebase pass, 2026-08-09)

**Rebase**: branch rebased onto today's main (8b3634f, 21 commits ahead of the previous base) with zero conflicts. Supersession check: no commits on main since the previous base touched the attack-path services, inject services, or the PayloadType/NetworkTraffic model, so nothing in this PR was superseded. Pushed with --force-with-lease; new head [5245f91](https://github.com/OpenAEV-Platform/openaev/commit/5245f91ab). All three commits signed.

**New review thread (Copilot balanced review, blocking)**: valid catch - a TargetedAsset argument is a JSON array of endpoint ids, so the scalar override read was blank and fell back to the payload default (behavior inherited from before this PR). Fixed by routing the argument through the same resolver execution uses (InjectService#retrieveValuesOfTargetedAssetFromInject), so the row carries the property value (hostname / seen IP / local IP) the command actually ran with. A selection resolving to several destinations stays ambiguous under the one-row-per-(inject, agent) model - getExecutionIndex must recompute the persisted id from that pair alone for Phase-B verdicts/traces/collectors - and falls back to the executing endpoint; fanning out to one row per destination is an explicit follow-up requiring a multi-valued Phase-B index. Thread replied and resolved; zero unresolved threads remain.

**Copilot naming nitpicks**: the tests introduced by this PR now follow the given_X_should_Y convention from testing.instructions.md.

**Tests**: 2 added (targeted-asset single destination resolves to the endpoint's property value; multi-endpoint selection falls back to the executing endpoint), both asserting index-vs-write id agreement. Full class green locally: 18/18.

**CI**: fully green on [5245f91](https://github.com/OpenAEV-Platform/openaev/commit/5245f91ab) - pipeline gate, all API test shards, full E2E matrix, Spotless, CodeQL, signed-commits, coverage upload. PR is mergeable/CLEAN; title and description updated to reflect the final state.

**Remaining blocker**: the changes-requested review from @EvaE-Filigran is the only thing preventing merge - it needs her re-review (or a maintainer dismissal) to lift. The NETWORK_TRAFFIC evidence (type, entity, and creation path all present on today's main; #1191 tracks the never-built executor capability) still stands and is unchanged by the rebase.

**Comment by EvaE-Filigran** [comment]
Payload not displayed fix by this PR https://github.com/OpenAEV-Platform/openaev/pull/7283
